File: src/main/scala/temportalist/compression/main/common/item/ItemCompressed.scala
package temportalist.compression.main.common.item

import java.util

import net.minecraft.creativetab.CreativeTabs
import net.minecraft.init.Items
import net.minecraft.item.{Item, ItemStack}
import temportalist.compression.main.common.Compression
import temportalist.compression.main.common.init.{Compressed, TabCompressed}
import temportalist.compression.main.common.lib.EnumTier
import temportalist.origin.api.common.item.ItemBase

/**
  *
  * Created by TheTemportalist on 4/14/2016.
  *
  * @author TheTemportalist
  */
class ItemCompressed extends ItemBase(Compression) with ICompressed {

	TabCompressed.add(this)

	override def getSubItems(itemIn: Item, tab: CreativeTabs, list: util.List[ItemStack]): Unit = {
		for {
			sample <- Seq[ItemStack](
				new ItemStack(Items.APPLE),
				new ItemStack(Items.POTATO),
				new ItemStack(Items.BOOK),
				new ItemStack(Items.BOAT)
			)
			tier <- EnumTier.values()
		} {
			list.add(Compressed.create(sample, tier = tier))
		}
	}

}

File: src/main/java/com/temportalist/compression/common/items/ItemCompressed.java
package com.temportalist.compression.common.items;

import com.temportalist.compression.client.ModelLoaderCompressed;
import com.temportalist.compression.common.init.CompressedStack;
import com.temportalist.compression.common.init.ModItems;
import com.temportalist.compression.common.lib.EnumTier;
import net.minecraft.client.util.ITooltipFlag;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.entity.Entity;
import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.NonNullList;
import net.minecraft.world.World;
import net.minecraftforge.client.model.ModelLoader;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

import javax.annotation.Nullable;
import java.util.List;

public class ItemCompressed extends ItemBase implements ICompressed {

    public static final NonNullList<ItemStack> SUBTYPES = NonNullList.create();

    public ItemCompressed() {
        super("compressedItem");
    }

    @Override
    public void registerModel()
    {
        ModelLoader.setCustomModelResourceLocation(this, 0, ModelLoaderCompressed.fakeRL);
    }

    @Override
    public void getSubItems(CreativeTabs tab, NonNullList<ItemStack> items) {
        items.addAll(SUBTYPES);
    }

    @Override
    public String getItemStackDisplayName(ItemStack stack) {
        return this.getDisplayName(stack);
    }

    @Override
    @SideOnly(Side.CLIENT)
    public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
        this.addInfo(stack, tooltip);
    }

    @Override
    public boolean hasCustomEntity(ItemStack stack) {
        return this.hasEntityCompressed();
    }

    @Nullable
    @Override
    public Entity createEntity(World world, Entity location, ItemStack itemstack) {
        return this.createEntityCompressed(world, location, itemstack);
    }

    @Override
    public void onUpdate(ItemStack stack, World worldIn, Entity entityIn, int itemSlot, boolean isSelected) {
        this.onUpdate(worldIn, entityIn, stack);
    }

    @Nullable
    @Override
    public NBTTagCompound getNBTShareTag(ItemStack stack)
    {
        return CompressedStack.getMinimizedTag(stack);
    }

}
